### ISSUE TITLE
Dereferences in property initializers don't update nullable state of dereferneced fields\properties

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#34868]
**Version Used**: 

Branch master (9 Apr 2019)
Latest commit 665dc2 by Andrew Hall:
Merge pull request #34764 from ryzngard/issue/pullmembers_up_a11y

Pull Members Up UI fixes

**Steps to Reproduce**:

Compile the following code:
```cs
#nullable enable
class C
{
  C? f = GetSomeValue();
  void M1()
  {
    var c = new C() { f = { f = new C() } }; // CS8602
    c.ToString();
    c.f.ToString(); // CS8602
  }
    
  public static extern C? GetSomeValue();
}
```

https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQBMQGoA+BiAdgVwBtCJhC4ACOXU8gWACgABAJgoGFGBvRijgfgoAzCgF4KAcTgwAygHswcAGoRC+OAAoAlAG5eFJgBYKAWQCM2/TwZ8+ANwgIKAYzEVccAO4dtFLsLd/EXEPb3ZfAF8KCL0bW2cAOgAVORkYBABLXABzbVjbFwShZNT0rNzdCgB6Ko4ZAA4ANgAGFn0I/T59JgBmAzNGqgAPeARcAUlpeUUVNU1dRgigA==

**Expected Behavior**:
A single CS8602 warning for the property initializer `f = { f = new C() }`

**Actual Behavior**:
There are two CS8602 warnings. One for the property initializer `f = { f = new C() }`, another for the `c.f.ToString` derefernece of the field `f`

**Notes**
The dereference of the field `f` in the property initializer should've updated the nullable state of the field `f` in `new C().f` object creation expression just as other kinds of dereference do. Consequently `c.f.ToString` should not issue a warning because the program reaches that point iff the field was initialized with non-nullable value. Otherwise it would've crashed at the object creation expression.
